specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2020-12-01-preview/costmanagement.generatedetailedcostreport.json

```diff
@@ -0,0 +1,405 @@
+          "type": "string"
+        },
+        "customerId": {
+          "description": "Customer Id for Modern (Invoice Id and billing profile is also required for this).",
+          "type": "string"
+        }
+      }
+    },
+    "GenerateDetailedCostReportTimePeriod": {
+      "description": "The start and end date for pulling data for the cost detailed report.",
+      "properties": {
+        "start": {
+          "description": "The start date to pull data from. example format 2020-03-15",
+          "type": "string"
+        },
+        "end": {
+          "description": "The end date to pull data to. example format 2020-03-15",
+          "type": "string"
+        }
+      },
+      "required": [
+        "start",
+        "end"
+      ]
+    },
+    "GenerateDetailedCostReportOperationStatuses": {
+      "description": "The status of the long running operation for cost detailed report.",
+      "properties": {
+        "id": {
+          "description": "The id of the long running operation.",
+          "type": "string"
+        },
+        "name": {
+          "description": "The name of the long running operation.",
+          "type": "string"
+        },
+        "status": {
+          "description": "The status of the long running operation.",
+          "$ref": "#/definitions/Status"
+        },
+        "type": {
+          "description": "The type of the long running operation.",
+          "type": "string"
+        },
+        "error": {
+          "description": "The details of the error.",
+          "$ref": "#/definitions/ErrorDetails"
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "type": "object",
+          "description": "The properties of the resource generated.",
+          "$ref": "#/definitions/DownloadURL"
+        }
+      }
+    },
+    "GenerateDetailedCostReportOperationResult": {
+      "description": "The result of the long running operation for cost detailed report.",
+      "properties": {
+        "id": {
+          "description": "The id of the long running operation.",
+          "type": "string"
+        },
+        "name": {
+          "description": "The name of the long running operation.",
+          "type": "string"
+        },
+        "type": {
+          "description": "The type of the long running operation.",
+          "type": "string"
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "type": "object",
+          "description": "The properties of the resource generated.",
+          "$ref": "#/definitions/DownloadURL"
+        }
+      }
+    },
+    "DownloadURL": {
+      "description": "The URL to download the generated report.",
+      "properties": {
+        "downloadUrl": {
+          "description": "The URL to download the generated report.",
+          "type": "string"
+        },
+        "validTill": {
```

Comment:
perhaps consider a different name such as expirationDate, also might be worth adding the format as part of the description: e.g. 2020-10-01 01:01 UTC something like that.

Reply:
Added more information in description